src/components/LibraryItem.tsx

```diff
@@ -249,6 +251,9 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
     onLibraryItemClicked() {
         // Toggle expansion state.
         let currentlyExpanded = this.state.expanded;
+        if (this.props.data.childItems.length > 0 && !currentlyExpanded) {
+            this.props.onExpand();
+        }
```

Comment:
The notification is sent before expansion actually take place, so let's rename this from `onExpand` to `onItemWillExpand` (just to borrow from React.js term `componentWillReceiveProps` and `componentWillUnmount`, etc).